### PR TITLE
Added new rego rules for Azure CSPM and fixed Issue #403

### PR DESCRIPTION
**Pull request by rezoan**

### CODE REVIEW

File: azure/cloud/master-compliance-test.json [hunk 1]
```diff
@@ -2771,7 +2771,7 @@
                         {
                             "cloud": "azure",
                             "compliance": [],
```
Comment (farchide) [inline azure/cloud/master-compliance-test.json:2773]: Please fill out compliance tags

File: azure/cloud/master-compliance-test.json [hunk 2]
```diff
@@ -2872,6 +2872,126 @@
...
+                            "remediationFunction": "PR_AZR_CLD_SQL_017.py"
+                        }
+                    ],
+                    "tags": [
+                        {
+                            "cloud": "azure",
+                            "compliance": [
```
Comment (farchide) [inline azure/cloud/master-compliance-test.json:2891]: compliance tags missing

File: azure/cloud/master-compliance-test.json [hunk 3]
```diff
@@ -2872,6 +2872,126 @@
...
+                            "eval": "data.rule.dbsec_threat_email",
+                            "message": "data.rule.dbsec_threat_email_err",
+                            "remediationDescription": "Login to Azure Portal. Go to SQL Databases (Left Panel). Choose the reported each DB instance. Click on Auditing & Threat Detection (under 'SETTINGS'). Set Auditing to 'ON'. Set Threat Detection to 'ON'. Set 'Send alerts to' field with a valid email address. Click on save",
+                            "remediationFunction": "PR_AZR_CLD_SQL_019.py"
+                        }
+                    ],
+                    "tags": [
+                        {
+                            "cloud": "azure",
+                            "compliance": [
```
Comment (farchide) [inline azure/cloud/master-compliance-test.json:2921]: compliance tags missing

File: azure/cloud/master-compliance-test.json [hunk 4]
```diff
@@ -2872,6 +2872,126 @@
...
+                            "eval": "data.rule.dbsec_threat_alert",
+                            "message": "data.rule.dbsec_threat_alert_err",
+                            "remediationDescription": "Login to Azure Portal. Go to SQL databases. For each DB instance. Click on Auditing & Threat Detection. Set Threat Detection types to All. Note: In order to Set Threat Detection types to All, Make sure Auditing is set to On.",
+                            "remediationFunction": "PR_AZR_CLD_SQL_020.py"
+                        }
+                    ],
+                    "tags": [
+                        {
+                            "cloud": "azure",
+                            "compliance": [
```
Comment (farchide) [inline azure/cloud/master-compliance-test.json:2951]: compliance tags missing

**Review by farchide — CHANGES_REQUESTED** [review]

File: azure/cloud/master-compliance-test.json (hunk 2 above)
Comment (rezoan) [reply azure/cloud/master-compliance-test.json:2949]: Added

File: azure/cloud/master-compliance-test.json (hunk 3 above)
Comment (rezoan) [reply azure/cloud/master-compliance-test.json:3061]: Added

File: azure/cloud/master-compliance-test.json (hunk 4 above)
Comment (rezoan) [reply azure/cloud/master-compliance-test.json:3177]: Added

File: azure/iac/web.rego [hunk 5]
```diff
@@ -122,45 +122,49 @@ min_tls_version {
...
-} else = "microsoft.web/sites resource property config.minTlsVersion is missing from the resource" {
-    lower(input.resources[_].type) == "microsoft.web/sites"
-    azure_attribute_absence["min_tls_version"]
-    azure_inner_attribute_absence["min_tls_version"]
 } else = "microsoft.web/sites resource property config.minTlsVersion is missing from the resource" {
     lower(input.resources[_].type) == "microsoft.web/sites"
     azure_inner_attribute_absence["min_tls_version"]
+    not azure_attribute_absence["min_tls_version"]
     azure_issue["min_tls_version"]
 } else = "microsoft.web/sites resource property config.minTlsVersion is missing from the resource" {
```
[the remark this answers is not in the document]
Comment (rezoan) [reply azure/iac/web.rego]: @shahinshirvani updated the message. please verify.

File: azure/cloud/dbsecurityalertpolicies.rego [hunk 6]
```diff
@@ -128,185 +118,153 @@ package rule
...
-#     "Language": "",
-#     "Policy Title": "Threat Detection alert should be configured to send notifications to the SQL server account administrators",
-#     "Policy Description": "Ensure that threat detection alert is configured to send notifications to the sql server account administrators",
-#     "Resource Type": "microsoft.sql/servers/databases/securityalertpolicies",
-#     "Policy Help URL": "",
-#     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/2018-06-01-preview/servers/databases/securityalertpolicies"
-# }
+default sql_alert = null
+
+azure_issue["sql_alert"] {
```
Comment (shahinshirvani) [inline azure/cloud/dbsecurityalertpolicies.rego:226]: @rezoan please change ```azure_issue``` to ```azure_attribute_absence```
also please do it for IaC as I checked

Comment (shahinshirvani) [reply azure/cloud/dbsecurityalertpolicies.rego:226]: All other changes were fine as I checked

Comment (rezoan) [reply azure/cloud/dbsecurityalertpolicies.rego]: @shahinshirvani done.

**Review by shahinshirvani — APPROVED** [review]

File: azure/iac/dbsecurityalertpolicies.rego [hunk 7]
```diff
@@ -231,7 +231,7 @@ dbsec_threat_alert_metadata := {
     "Type": "IaC",
     "Product": "AZR",
     "Language": "ARM template",
-    "Policy Title": "Azure SQL Server threat detection alerts should be enabled for all threat types",
+    "Policy Title": "Azure SQL databse threat detection alerts should be enabled for all threat types",
```
Comment (farchide) [inline azure/iac/dbsecurityalertpolicies.rego:234]: typo: `database`

Comment (rezoan) [reply azure/iac/dbsecurityalertpolicies.rego]: @farchide fixed.

**Review by farchide — APPROVED** [review]

**Pull request merged**